compiler/rustc_codegen_llvm/src/consts.rs

```diff
@@ -96,10 +97,14 @@ pub fn const_alloc_to_llvm<'ll>(cx: &CodegenCx<'ll, '_>, alloc: ConstAllocation<
 
         llvals.push(cx.scalar_to_backend(
             InterpScalar::from_pointer(Pointer::new(prov, Size::from_bytes(ptr_offset)), &cx.tcx),
-            Scalar::Initialized {
-                value: Primitive::Pointer(address_space),
-                valid_range: WrappingRange::full(dl.pointer_size),
-            },
+            cx.tcx.mk_layout(LayoutS::scalar(
+                cx,
+                Scalar::Initialized {
+                    value: Primitive::Pointer(address_space),
+                    valid_range: WrappingRange::full(dl.pointer_size),
+                },
+                ReprOptions::default(),
```

Comment:
This is suspicious and almost certainly wrong. Not sure how to construct or retrieve the correct representation options here.